client/sites/hooks/use-transfer-with-software-start-mutation.ts

```diff
@@ -0,0 +1,48 @@
+import { useMutation, UseMutationResult } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+type transferWithSoftwareResponse = {
+	transferId: number;
+};
+
+const requestTransferWithSoftware: (
+	siteId: number,
+	from?: string,
+	plugins?: Record< string, 'install' | 'activate' >,
+	themes?: Record< string, 'install' | 'activate' >
+) => Promise< transferWithSoftwareResponse > = async (
+	siteId: number,
+	from?: string,
+	plugins?: Record< string, 'install' | 'activate' >,
+	themes?: Record< string, 'install' | 'activate' >
+) => {
+	const response = await wpcom.req.post(
+		{
+			path: `/sites/${ siteId }/atomic/transfer-with-software?http_envelope=1`,
+		},
+		{
+			apiNamespace: 'wpcom/v2',
+			body: { plugins, themes, migration_source_site_domain: from },
+		}
+	);
+
+	if ( ! response ) {
+		throw new Error( 'Transfer with software failed' );
+	}
+
+	return response;
+};
+
+export const useRequestTransferWithSoftware = (
+	siteId?: number,
+	from?: string,
+	plugins?: Record< string, 'install' | 'activate' >,
```

Comment:
```suggestion
type PluginSlug = string
type PluginState = 'install' | 'activate' 
....
	plugins?: Record< PluginSlug, 'install' | 'activate' >,
```

The idea here is to describe better what we are expecting